app.py

```diff
@@ -1,17 +1,23 @@
 #!/usr/bin/env python3
 import os
 import boto3 
+import string
+import random
 import aws_cdk as cdk
 import aws_cdk.assertions as assertions
 from sds_in_a_box.sds_in_a_box_stack import SdsInABoxStack
 
-
+### Implement the code found in "sds_in_a_box_stack.py"
 app = cdk.App()
-SdsInABoxStack(app, "SdsInABoxStack")
 
-app.synth()
+SDS_ID = app.node.try_get_context("SDSID")
+
+if SDS_ID is None:
+    raise Exception("ERROR: Need to specify an ID to name the stack (ex - production, testing, etc)")
```

Comment:
If it is None do you want to just assign it the random ID and log that out for the user, or warn and say you're using a random ID?

```suggestion
if SDS_ID is None:
    raise ValueError("the context SDS_ID needs to specify an ID to name the stack (ex - production, testing, etc)")
```

Reply:
I would probably just err on the side of forcing a user to specify the context, but I'm fine either way

Reply:
Sounds reasonable to me to force a user to provide one rather than default.